http/src/main/java/io/micronaut/http/reactive/execution/ReactorExecutionFlowImpl.java

```diff
@@ -102,7 +130,20 @@ public Context currentContext() {
             @Override
             public void onSubscribe(Subscription s) {
                 this.subscription = s;
-                s.request(1);
+                boolean cancel;
+                synchronized (ReactorExecutionFlowImpl.this) {
```

Comment:
I don't understand what is happening. Should we avoid a subscription if the flow is canceled?

Reply:
ExecutionFlow downstream APIs (onComplete and cancel) are generally not thread safe. My assumption with this API is that cancel is always called strictly after onComplete is called. In particular, DelayedFlow will throw an exception if you call onComplete after cancel. That means that if the flow is cancelled, the user shouldn't be subscribing afterwards anyway (though there is no check in ReactiveExecutionFlow).

This check is only necessary because onSubscribe can happen some time delayed after onComplete is called, so by that time the flow *can* be cancelled.